core/phase.js

```diff
@@ -113,7 +113,9 @@ Tags.prototype.read = function(key) {
 
 function PhaseBaseRuntime(base, impl) {
   this.phaseBase = base;
-  this.impl = trace.wrap({cat: 'core', name: base.name}, impl.bind(this));
+  this.impl = trace.wrap(function() {
+    return {cat: 'core', name: base.name, args: {tags: this.tags.tags}};
```

Comment:
I'm not sure this always makes sense. 0:1 arity phases won't have any tags, and for other phases this will only capture the tags for the first input.